utils/utils.js

```diff
@@ -1187,3 +1187,30 @@ export function enforceWhitespace(el, offset, direction, rule) {
         spaceNode.nodeValue = spaceNode.nodeValue.replace(expr, spaceVisibility ? '\u00A0' : '');
     }
 }
+/**
+ * Generates an iterable of every node of a range.
+ * 
+ * @param {Range} range 
+ */
+export function* getRangeRover(range) {
+    const iterator = document.createNodeIterator(
+        range.commonAncestorContainer,
+        NodeFilter.SHOW_ALL, // pre-filter
+        {
+            // custom filter
+            acceptNode: function (node) {
+                return NodeFilter.FILTER_ACCEPT;
+            }
+        }
+    );
+    let started = false;
+    while (iterator.nextNode()) {
+        if (!started && iterator.referenceNode !== range.startContainer) { continue; }
```

Comment:
je comprends pas le truc de started... c'est juste pour faire `iterator.nextNode()` jusqu'à ce que la deuxième condition soit true ? autant le faire en dehors non ? Parce que là c'est vraiment confusing je trouve. On peut pas juste initialiser l'iterator sur le bon noeud directement ? j'ai testé et si ça a l'air de marcher. Du coup j'ai réécris l'itérateur plus bas.

Reply:
Le node iterator itère sur tout les enfants du commonAncestor, mais je veux seulement ceux qui sont dans le range, donc j'attend de rencontrer le start container pour commencer à yielder.

Reply:
Je sais pas ce qui est mieux en fait.  Deux boucles simplissimes ou une boucle plus compliquée qui fait plusieurs trucs en même temps...

Donc la version actuelle vs:
```js
const iterator = document.createNodeIterator(range.commonAncestorContainer);
while (iterator.nextNode() && iterator.referenceNode !== range.startContainer);
do {
    yield currentNode;
} while (iterator.nextNode() && iterator.referenceNode !== range.endContainer);
```